SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1300,9 +1301,31 @@ void USpatialActorChannel::ServerProcessOwnershipChange()
 		}
 	}
 
-	UpdateEntityACLToNewOwner();
+	// Changing an Actor's owner can affect its NetConnection, so we need to reevaluate this.
+	FString NewClientConnectionWorkerId;
+	{
+		SCOPE_CYCLE_COUNTER(STAT_GetOwningClientWorkerId);
+		NewClientConnectionWorkerId = SpatialGDK::GetOwningClientWorkerId(Actor);
+	}
+
+	// Update the ComponentPresence component.
+	check(NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID));
+	SpatialGDK::ComponentPresence* ComponentPresenceData = NetDriver->StaticComponentView->GetComponentData<SpatialGDK::ComponentPresence>(EntityId);
+	ComponentPresenceData->PossessingClientWorkerId = NewClientConnectionWorkerId;
+	FWorkerComponentUpdate Update = ComponentPresenceData->CreateComponentPresenceUpdate();
+	NetDriver->Connection->SendComponentUpdate(EntityId, &Update);
+
+	// Update the EntityACL component (if authoritative).
+	if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
```

Comment:
If the server which is auth for the entity is not auth for the ACL, will it call ProcessOwnershipChange somehow because of the change to the OwningClientWorkerId? I didn't see that in the code, but I might be overlooking it.

Reply:
If the simulating worker doesn't have ACL auth, the ComponentPresence update is received by the enforcer, and from there it processes the changing to the UnrealClientEndpoint and Heartbeat components.

Reply:
Gotcha, thanks!